### PR TITLE
libgpm 1.20.7 (new formula)

### PR DESCRIPTION
**Pull request by dochang**

### CODE REVIEW

File: Library/Formula/libgpm.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  depends_on "libtool" => :build
+  depends_on "bison" => :build
+  depends_on "texinfo" => :build
+  depends_on "homebrew/dupes/ncurses" unless OS.mac?
+
+  # https://github.com/telmich/gpm/pull/14
+  patch :DATA
+
+  def install
+    ENV.deparallelize
```
Comment (sjackman) [inline Library/Formula/libgpm.rb]: Please report the error message that you see without `ENV.deparallelize`

File: Library/Formula/libgpm.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
+  depends_on "bison" => :build
+  depends_on "texinfo" => :build
+  depends_on "homebrew/dupes/ncurses" unless OS.mac?
+
+  # https://github.com/telmich/gpm/pull/14
+  patch :DATA
```
Comment (sjackman) [inline Library/Formula/libgpm.rb]: Please use
```ruby
patch do
  url "https://patch-diff.githubusercontent.com/raw/telmich/gpm/pull/14.patch"
  sha256 "fix me"
end
```

File: Library/Formula/libgpm.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
+class Libgpm < Formula
+  desc "general purpose mouse"
+  homepage "http://www.nico.schottelius.org/software/gpm/"
+  url "http://www.nico.schottelius.org/software/gpm/archives/gpm-1.20.7.tar.bz2"
+  sha256 "f011b7dc7afb824e0a017b89b7300514e772853ece7fc4ee640310889411a48d"
+
+  head "https://github.com/telmich/gpm.git"
```
Comment (sjackman) [inline Library/Formula/libgpm.rb]: Please add
```ruby
# tag "linuxbrew"
```

Comment (sjackman) [inline Library/Formula/libgpm.rb]: Unless this formula also compiles correctly on Mac OS and you can test it, in which case send it upstream to Homebrew-core!

**Comment by sjackman** [comment]
Nice work! Thanks, Desmond.

File: Library/Formula/libgpm.rb (hunk 1 above)
Comment (dochang) [inline Library/Formula/libgpm.rb]:
```
/usr/bin/gcc-5 -L/home/doc/.linuxbrew/lib -Wl,--dynamic-linker=/home/doc/.linuxbrew/lib/ld.so -Wl,-rpath,/home/doc/.linuxb
rew/lib -L/tmp/libgpm20160413-3016-1win64j/gpm-1.20.7/src -L/home/doc/.linuxbrew/lib -Wl,--dynamic-linker=/home/doc/.linux
brew/lib/ld.so -Wl,-rpath,/home/doc/.linuxbrew/lib -o prog/mouse-test prog/mouse-test.o mice.o twiddler.o synaptics.o prog
/open_console.o lib/libgpm.so.2   -lm
/usr/bin/gcc-5 -L/home/doc/.linuxbrew/lib -Wl,--dynamic-linker=/home/doc/.linuxbrew/lib/ld.so -Wl,-rpath,/home/doc/.linuxb
rew/lib -L/tmp/libgpm20160413-3016-1win64j/gpm-1.20.7/src -L/home/doc/.linuxbrew/lib -Wl,--dynamic-linker=/home/doc/.linux
brew/lib/ld.so -Wl,-rpath,/home/doc/.linuxbrew/lib -o prog/disable-paste prog/disable-paste.o lib/libgpm.so.2  
gcc-5: error: prog/mev.o: No such file or directory
Makefile:65: recipe for target 'prog/mev' failed
make[1]: *** [prog/mev] Error 1
make[1]: *** Waiting for unfinished jobs....
gcc-5: error: prog/hltest.o: No such file or directory
Makefile:65: recipe for target 'prog/hltest' failed
make[1]: *** [prog/hltest] Error 1
gcc-5: error: prog/disable-paste.o: No such file or directory
Makefile:65: recipe for target 'prog/disable-paste' failed
make[1]: *** [prog/disable-paste] Error 1
make[1]: Leaving directory '/tmp/libgpm20160413-3016-1win64j/gpm-1.20.7/src'
Makefile:77: recipe for target 'do-install' failed
make: *** [do-install] Error 1
```

**Comment by dochang** [comment]
All issues are fixed.  Thanks for your help!

**Comment by sjackman** [comment]
Do you have access to a Mac, and if so do you know whether it works on a Mac?

**Comment by sjackman** [comment]
No, it does not work on Mac.
```
mice.c:59:10: fatal error: 'linux/kdev_t.h' file not found
```
https://gist.github.com/sjackman/072b5c9d0d720192153e565112d77234#file-03-make-L15

**Comment by sjackman** [comment]
Merged. I manually force the bottle to `cellar :any` as it doesn't install any data.
Thanks for your contribution, Desmond!

**Pull request closed without merging**